Saucier720-app/src/app/list/list.component.ts

```diff
@@ -60,31 +60,73 @@ export class ListComponent implements OnInit {
     }
   }
 
-  addIngredient() {
-    if (this.newIngredientName && this.newIngredientQuantity > 0) {
-      const newIngredient: Ingredient = {
-        Name: this.newIngredientName,
-        Quantity: this.newIngredientQuantity,
+  // Optional parameter so that we can call it from deals and recipes pages 
+  addIngredient(ingredient?: Ingredient | string) {
+
+    // Check if ingredient exists in userList already
+    // create function validateIngredient 
+
+
+
+    // Ingredient is not in list 
+    let newIngredient: Ingredient | null = null; 
+    
+    // Assigns values from already created foodItem
+    if (typeof ingredient === 'string') {
+      newIngredient = {
+        Name: ingredient,
+        Quantity: 1,
         StoreCost: 0, // Example value, replace with actual value if needed
         OnSale: false, // Example value, replace with actual value if needed
         SalePrice: 0, // Example value, replace with actual value if needed
         SaleDetails: '' // Example value, replace with actual value if needed
       };
+    } else if(ingredient){
```

Comment:
Does this check if it's type ingredient, or just if it's null or not? If I'm following the logic it seems like it's checking the type, but wouldn't this syntax just check if it's null or not?

Reply:
This is checkin if a type ingredient was inputted/if it was null or not. We can't add to list if the 'quantity' is 0, so i felt like the easiest thing to do check the input and go down the rest of the function linearly as it was originally intended.

Reply:
Yes, so shouldn't this line check the type by using typeof or whatever

Reply:
In theory yes but because we are using 'typeof' with an interface type which can't exist at runtime. 
<img width="537" alt="image" src="https://github.com/TylerMetz/Saucier720/assets/92872780/0f26a872-f714-4838-9095-309b300bc05d">